src/ctap/mod.rs

```diff
@@ -330,20 +330,21 @@ impl StatefulPermission {
 
 // This struct currently holds all state, not only the persistent memory. The persistent members are
 // in the persistent store field.
-pub struct CtapState {
-    persistent_store: PersistentStore,
+pub struct CtapState<E: Env> {
+    persistent_store: PersistentStore<E>,
```

Comment:
The PersistentStore line is the reason why we have to introduce `E: Env` here (and in `Ctap`), right? I wonder if there is a way around this.

Reply:
Yes that's the reason.

The way around would be to have the persistent store part of the environment directly. Let me try to write a new PR with that idea (and include the fix for all open comments here that would still be applicable).

Reply:
I created #438 and would suggest merging that one instead of this one. I'll rebase once the one to be merged is identified to only rebase that one.